pkg/gadgets/filetop/tracer/tracer.go

```diff
@@ -0,0 +1,250 @@
+		t.readLink.Close()
+		t.readLink = nil
+	}
+	if t.writeLink != nil {
+		t.writeLink.Close()
+		t.writeLink = nil
+	}
+
+	t.objs.Close()
+}
+
+func (t *Tracer) start() error {
+	spec, err := loadFiletop()
+	if err != nil {
+		return fmt.Errorf("Failed to load ebpf program: %w", err)
+	}
+
+	filter_by_mnt_ns := false
+
+	if t.c.MountnsMap != "" {
+		filter_by_mnt_ns = true
+		m := spec.Maps["mount_ns_set"]
+		m.Pinning = ebpf.PinByName
+		m.Name = filepath.Base(t.c.MountnsMap)
+	}
+
+	consts := map[string]interface{}{
+		"target_pid":        uint32(t.c.TargetPid),
+		"regular_file_only": !t.c.AllFiles,
+		"filter_by_mnt_ns":  filter_by_mnt_ns,
+	}
+
+	if err := spec.RewriteConstants(consts); err != nil {
+		return fmt.Errorf("error RewriteConstants: %w", err)
+	}
+
+	opts := ebpf.CollectionOptions{
+		Maps: ebpf.MapOptions{
+			PinPath: filepath.Dir(t.c.MountnsMap),
+		},
+	}
+
+	if err := spec.LoadAndAssign(&t.objs, &opts); err != nil {
+		return fmt.Errorf("Failed to load ebpf program: %w", err)
+	}
+
+	kpread, err := link.Kprobe("vfs_read", t.objs.VfsReadEntry)
+	if err != nil {
+		return fmt.Errorf("Error opening kprobe: %w", err)
+	}
+	t.readLink = kpread
+
+	kpwrite, err := link.Kprobe("vfs_write", t.objs.VfsWriteEntry)
+	if err != nil {
+		return fmt.Errorf("Error opening kprobe: %w", err)
+	}
+	t.writeLink = kpwrite
+
+	t.run()
+
+	return nil
+}
+
+func (t *Tracer) next_stats() ([]types.Stats, error) {
+	stats := []types.Stats{}
+
+	var prev *C.struct_file_id = nil
+	key := C.struct_file_id{}
+	entries := t.objs.Entries
+
+	defer func() {
+		// delete elements
+		err := entries.NextKey(nil, unsafe.Pointer(&key))
+		if err != nil {
+			return
+		}
+
+		for {
+			if err := entries.Delete(key); err != nil {
+				return
+			}
+
+			prev = &key
+			if err := entries.NextKey(unsafe.Pointer(prev), unsafe.Pointer(&key)); err != nil {
+				return
+			}
+		}
+	}()
+
+	// gather elements
+	err := entries.NextKey(nil, unsafe.Pointer(&key))
+	if err != nil {
+		if errors.Is(err, ebpf.ErrKeyNotExist) {
+			return stats, nil
+		}
+		return nil, fmt.Errorf("Error getting next key: %w", err)
+	}
+
+	for {
+		file_stat := C.struct_file_stat{}
+		if err := entries.Lookup(key, unsafe.Pointer(&file_stat)); err != nil {
+			return nil, err
+		}
+
+		stat := types.Stats{
+			Reads:      uint64(file_stat.reads),
+			Writes:     uint64(file_stat.writes),
+			ReadBytes:  uint64(file_stat.read_bytes),
+			WriteBytes: uint64(file_stat.write_bytes),
+			Pid:        uint32(file_stat.pid),
+			Tid:        uint32(file_stat.tid),
+			Filename:   C.GoString(&file_stat.filename[0]),
+			Comm:       C.GoString(&file_stat.comm[0]),
+			FileType:   byte(file_stat.type_),
+			MountNsId:  uint64(file_stat.mntns_id),
+		}
+
+		container := t.resolver.LookupContainerByMntns(stat.MountNsId)
+		if container != nil {
+			stat.Container = container.Name
+			stat.Pod = container.Podname
+			stat.Namespace = container.Namespace
+			stat.Node = os.Getenv("NODE_NAME")
```

Comment:
Why are you getting the node name like this instead of retrieving it from the trace specs (`trace.Spec.Node`) as the other gadgets do? 

What about sending it in the `filetoptracer.Config`?

```suggestion
			stat.Node = t.c.NodeName
```

Reply:
There is not any particular reason. Using the env was the first idea I have. I agree it's better to use `trace.Spec.Node`.